ChangeLog.md

```diff
@@ -0,0 +1,5 @@
+# Change log
```

Comment:
Not sure if this PR is a good starting point for a changelog. I can see myself being too lazy to maintain one. @gdmcbain What do you think?

Reply:
I don't use change logs myself. Do they add anything over `git log`, `git blame`, &c.?

Reply:
The main objective is to inform users of (major) API changes...

Reply:
I got to say that I always find it very convenient to have changelog in other pieces of software since I'm curious about new things and too lazy to dig through the git history of other projects.

For us to maintain properly, I think it's too much work atm. Let's not start this as part of this PR.

Reply:
Yes maybe for this (currently relatively not so big) project git history is enough..

Reply:
Let's remove the changelog then.